### DISCUSSION TITLE
How to add MS SQL Server persistence to dispatcher provider?

### DISCUSSION THREAD

**Discussion by tsoG2000** [elsa-workflows/elsa-core#1616] [Q&A] [answered]
We are using default Elsa DispatchAsync(). How can we configure it to use MS SQL Server instead of InMemoryTransport?

**Comment by sfmskywalker** [comment, marked as answer]
`DispatchAsync` uses the Rebus by default to publish messages to the configured message broker. You can configure the MS SQL Server broker for Rebus like this:

```csharp
services.AddElsa(elsa =>
{
   elsa.UseServiceBus(bus =>
    {
        bus.Configurer.Transport(transport => transport.UseSqlServer(new SqlServerTransportOptions(sqlConnnectionString), bus.QueueName));
    });
});
```

Be sure to provide the queue name that you receive via the `bus` argument (as shown in above code snippet).
